File: cs/Day7.cs
using System;
using System.IO;
using System.Collections.Generic;

class Day7 {
  (T?, T?) MinMax<T>(IEnumerable<T> elems) {
    return (elems.Min(), elems.Max());
  }

  public void solve() {
    var data = File.ReadAllText("../input7.txt").Trim().Split(',').Select(long.Parse);
    var (l, h) = MinMax(data);
    long minFuel = 999_999_999;
    for (long i = l; i <= h; ++i) {
      long fuel = 0;
      foreach (var crab in data) {
        for (long step = 1; step <= Math.Abs(crab-i); ++step) {
          fuel += step;
        }
      }
      minFuel = Math.Min(fuel, minFuel);
    }
    Console.WriteLine(minFuel);
  }
}

File: day7.nim
import std/[strutils, strscans, sequtils, sugar]

proc minmax[T](arr: openArray[T]): (T, T) =
    return (arr.min, arr.max)

proc solve(data: seq[int]): int =
    let (l, h) = minmax(data)
    var minFuel = 999_999_999
    for i in l..h:
        var fuel = 0
        for crab in data:
            for step in 1..abs(crab-i):
                fuel += step
        minFuel = min(fuel, minFuel)
    return minFuel

let data = readFile("input7.txt").strip(leading=false).split(',').map(parseInt)
echo data.solve